src/navigator-card-body.jsx

```diff
@@ -0,0 +1,302 @@
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "za":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.name.toLowerCase() > b.name.toLowerCase()
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "last_modified":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.mtime > b.mtime
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "first_modified":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.mtime < b.mtime
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    default:
+        break;
+    }
+};
+
+export const NavigatorCardBody = ({
+    currentFilter,
+    files,
+    historyIndex,
+    isGrid,
+    path,
+    selected,
+    setHistory,
+    setHistoryIndex,
+    setSelected,
+    setSelectedContext,
+    sortBy,
+    loadingFiles
+}) => {
+    const [boxPerRow, setBoxPerRow] = useState(0);
+    const Dialogs = useDialogs();
+    const sortedFiles = useMemo(() => {
+        const compareFunc = compare(sortBy);
+
+        return files
+                .filter(file => {
+                    return file.name.toLowerCase().includes(currentFilter.toLowerCase());
+                })
+                .sort(compareFunc);
+    }, [files, currentFilter, sortBy]);
+    const isMounted = useRef(null);
+
+    function calculateBoxPerRow () {
+        const boxes = document.querySelectorAll(".item-button");
+        if (boxes.length > 1) {
+            let i = 0;
+            const total = boxes.length;
+            const firstOffset = boxes[0].offsetTop;
+            while (++i < total && boxes[i].offsetTop === firstOffset);
+            setBoxPerRow(i);
+        }
+    }
+
+    const onDoubleClickNavigate = useCallback((file) => {
+        const newPath = [...path, file.name].join("/");
+        if (file.type === "dir" || file.to === "dir") {
+            setHistory(h => [...h.slice(0, historyIndex + 1), [...path, file.name]]);
+            setHistoryIndex(h => h + 1);
+
+            cockpit.location.go("/", { path: encodeURIComponent(newPath) });
+        }
+    }, [
+        path,
+        historyIndex,
+        setHistoryIndex,
+        setHistory
+    ]);
+
+    useEffect(() => {
+        calculateBoxPerRow();
+        window.onresize = calculateBoxPerRow;
+        return () => {
+            window.onresize = undefined;
+        };
+    });
+
+    useEffect(() => {
+        const onKeyboardNav = (e) => {
+            if (e.key === "ArrowRight") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = selectedIdx < sortedFiles.length - 1
+                        ? selectedIdx + 1
+                        : 0;
```

Comment:
These 7 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)